core/src/main/java/hudson/FilePath.java

```diff
@@ -3331,7 +3310,7 @@ public Boolean invoke(@Nonnull File parentFile, @Nonnull VirtualChannel channel)
                         Path child = currentFileAbsolutePath.normalize();
                         Path parent = parentAbsolutePath.normalize();
                         return child.startsWith(parent);
-                    } catch (InvalidPathException e2) {
+                    } catch (InvalidPathException e2) { // TODO which method would throw this?
```

Comment:
Undocumented for all of them, so remove? Or are we just extra careful here?

Reply:
It could probably be removed. I was erring on the side of caution. This `catch` originates in @Wadeck’s https://github.com/jenkinsci/jenkins/commit/c19cc705688cfffa4fe735e0edbe84862b6c135f. Indeed there is no documented throw of `InvalidPathException`, and from inspecting `UnixPath` and `WindowsPath` implementations of `normalize()` and `startsWith(Path)` I could find no case where it would be thrown; the closest I got was [this](http://hg.openjdk.java.net/jdk8u/jdk8u/jdk/file/f0b93fbd8cf8/src/windows/classes/sun/nio/fs/WindowsPath.java#l120), demonstrating that the authors considered this sort of thing.

Reply:
+1 for removal, it was surely added when I was using `File.toPath` but the code was finally using `Path` directly. Especially in the previous version of that PR, we were targeting an older version of the core, where the `Util.fileToPath` did not exist.